packages/fxa-settings/src/pages/Signin/InlineRecoverySetup/index.tsx

```diff
@@ -0,0 +1,225 @@
+import { Localized } from '@fluent/react';
+
+export type InlineRecoverySetupProps = {
+  isIos?: boolean;
+  recoveryCodes: Array<string>;
+  serviceName?: string;
+  showConfirmation: boolean;
+};
+
+const InlineRecoverySetup = ({
+  isIos,
+  recoveryCodes,
+  serviceName,
+  showConfirmation,
+}: InlineRecoverySetupProps & RouteComponentProps) => {
+  logPageViewEvent('inline-recovery-setup', {
+    entrypoint_variation: 'react',
+  });
+
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedInputTextLabel = ftlMsgResolver.getMsg(
+    'backup-authentication-code',
+    'Backup authentication code'
+  );
+  const localizedCopyButtonTitle = ftlMsgResolver.getMsg('copy-button', 'Copy');
+  const localizedDownloadButtonTitle = ftlMsgResolver.getMsg(
+    'download-button',
+    'Download'
+  );
+  const localizedPrintButtonTitle = ftlMsgResolver.getMsg(
+    'print-button',
+    'Print'
+  );
+  const localizedGraphicAriaLabel = ftlMsgResolver.getMsg(
+    'inline-recovery-setup-image-description',
+    'Document that contains hidden text.'
+  );
+
+  /* TODO: - Add in copy/download/print/continue/cancel actions for all buttons
+   *       - Add in metrics for all events
+   *       - Add tests for all metrics
+   */
+  const printCode = () => {};
+  const copyCode = () => {};
+  const downloadCode = () => {};
+  const cancelSetup = () => {};
+  const continueSetup = () => {};
+
+  return (
+    <>
+      {showConfirmation ? (
+        <>
+          <div className="mb-4">
+            {serviceName ? (
+              <FtlMsg id="inline-recovery-confirmation-header">
+                <h1 className="card-header">
+                  Confirm backup authentication code{' '}
+                  <span className="card-subheader">{`to continue to ${serviceName}`}</span>
+                </h1>
+              </FtlMsg>
+            ) : (
+              <FtlMsg id="inline-recovery-confirmation-header-default">
+                <h1 className="card-header">
+                  Confirm backup authentication code{' '}
+                  <span className="card-subheader">
+                    to continue to account settings
+                  </span>
+                </h1>
+              </FtlMsg>
+            )}
+          </div>
+          <section>
+            <form noValidate>
+              <div>
+                <RecoveryCodesGraphic
+                  role="img"
+                  aria-label={localizedGraphicAriaLabel}
+                  className="mx-auto"
+                />
+                <FtlMsg id="inline-recovery-confirmation-description">
+                  <p className="text-sm mb-6">
+                    To ensure that you will be able to regain access to your
+                    account, in the event of a lost device, please enter one of
+                    your saved backup authentication codes.
+                  </p>
+                </FtlMsg>
+                <div className="input-row">
```

Comment:
This div might not be needed? I'm not sure the input and buttons should all be within one input-row div (I might be missing where it's needed for certain screen sizes though!)

Reply:
No I think you're right, this is probably just vestigial